### DISCUSSION TITLE
Run tflint on *.hcl files

### DISCUSSION THREAD

**Discussion by gdsotirov** [terraform-linters/tflint#1912] [Ideas]
## Introduction

Tflint is capable of verifying `*.hcl` files (e.g. Terragrunt configurations), but currently I have to rename these to `*.tf`, which is inconvenient. It would be nice if I'm able to run `tflint` also on `*.hcl` files with some argument.

## Proposal

Allow checking of `*.hcl` files with an argument (e.g. `--hcl` or `--check-hcl`).

**Comment by bendrucker** [comment]
See https://github.com/terraform-linters/tflint/pull/1860, particularly my latest comment. We're not moving forward with that proposal to just hack on handling based on the `.hcl` extension. But there is potentially room for a new plugin API for loading arbitrary HCL files. Although at that point, it's probably just a better idea to provide an API for loading entirely arbitrary files as `[]byte`.  Plugin authors can just parse the HCL themselves.

**Reply by gdsotirov** [reply]
Thanks for referring #1860, because it seems to address exactly the raw need. My point is that just renaming `.hcl` to `.tf` basically works - I've cleansed many Terragrunt configurations like this. Anyway, I understand your reasons, because although both `.hcl` and `.tf` are using [the same syntax](https://terragrunt.gruntwork.io/docs/getting-started/configuration/) (of [HashiCorp/Terraform configuration language](https://developer.hashicorp.com/terraform/language)) their structure (blocks and attributes) are not the same and `.hcl` files do not compose Terraform modules. Let's hope a plugin for `.hcl` files would appear in future. :-)
